integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -113,7 +114,29 @@ fn test_resharding_v3() {
         let prev_epoch_id =
             client.epoch_manager.get_prev_epoch_id_from_prev_block(&tip.prev_block_hash).unwrap();
         let epoch_config = client.epoch_manager.get_epoch_config(&prev_epoch_id).unwrap();
-        epoch_config.shard_layout.shard_ids().count() == expected_num_shards
+        if epoch_config.shard_layout.shard_ids().count() != expected_num_shards {
+            return false;
+        }
+
+        // If resharding happened, also check that each shard has non-empty state.
+        for shard_id in 0..3 {
```

Comment:
nit: `in shard_layout.shard_ids()` or `shard_uids()` (replace shard layout with the right one)

Reply:
Done in #12221